diff -Naurp pam_keystore-0.1.3/pam_keystore.c.orig pam_keystore-0.1.3/pam_keystore.c
--- pam_keystore-0.1.3/pam_keystore.c.orig	2015-01-17 19:59:10.653505879 +0100
+++ pam_keystore-0.1.3/pam_keystore.c	2015-01-17 23:32:30.547010773 +0100
@@ -67,7 +67,7 @@ if (retval!=PAM_SUCCESS) {return PAM_AUT
 int flag = 0;
      for (ctrl=0; argc-- > 0; ++argv) 
      {
-       if (!strcmp(*argv,"use_first_pass"))
+       if (!strcmp(argv[ctrl],"use_first_pass"))
        {
          retval = pam_get_item(pamh,PAM_AUTHTOK,(const void **)&p);
          if (retval!=PAM_SUCCESS) {return PAM_AUTH_ERR;};
@@ -104,6 +104,7 @@ if (strcmp(name,"root")){
 
     // записываем ключ в пространство root
     ret = add_key("user", name, p, strlen(p), dest);
+    printf("Added key for user: %s\n", name);
     };
   }
 
