--- midori-0.5.5.orig/midori/midori-preferences.c	2013-08-14 20:32:02.000000000 +0200
+++ midori-0.5.5/midori/midori-preferences.c	2015-12-03 14:22:12.521410218 +0100
@@ -513,46 +513,46 @@
     /* Page "Network" */
     PAGE_NEW (GTK_STOCK_NETWORK, _("Network"));
     FRAME_NEW (NULL);
-    label = gtk_label_new (_("Proxy server"));
-    gtk_misc_set_alignment (GTK_MISC (label), 0.0, 0.5);
-    INDENTED_ADD (label);
-    button = katze_property_proxy (settings, "proxy-type", NULL);
-    SPANNED_ADD (button);
-    label = gtk_label_new (_("Hostname"));
-    gtk_misc_set_alignment (GTK_MISC (label), 0.0, 0.5);
-    INDENTED_ADD (label);
-    entry = katze_property_proxy (settings, "http-proxy", "address");
-    SPANNED_ADD (entry);
-    g_signal_connect (settings, "notify::proxy-type",
-        G_CALLBACK (midori_preferences_notify_proxy_type_cb), entry);
-    midori_preferences_notify_proxy_type_cb (settings, NULL, entry);
-    label = gtk_label_new (_("Port"));
-    gtk_misc_set_alignment (GTK_MISC (label), 0.0, 0.5);
-    INDENTED_ADD (label);
-    entry = katze_property_proxy (settings, "http-proxy-port", NULL);
-    SPANNED_ADD (entry);
-    g_signal_connect (settings, "notify::proxy-type",
-        G_CALLBACK (midori_preferences_notify_proxy_type_cb), entry);
-    midori_preferences_notify_proxy_type_cb (settings, NULL, entry);
-    INDENTED_ADD (gtk_event_box_new ());
-    label = gtk_label_new (NULL);
-    GString* proxy_types = g_string_new (NULL);
-    g_string_append (proxy_types, "<span size=\"smaller\">");
-    g_string_append (proxy_types, _("Supported proxy types:"));
-    g_string_append_c (proxy_types, ' ');
+    //label = gtk_label_new (_("Proxy server"));
+    //gtk_misc_set_alignment (GTK_MISC (label), 0.0, 0.5);
+    //INDENTED_ADD (label);
+    //button = katze_property_proxy (settings, "proxy-type", NULL);
+    //SPANNED_ADD (button);
+    //label = gtk_label_new (_("Hostname"));
+    //gtk_misc_set_alignment (GTK_MISC (label), 0.0, 0.5);
+    //INDENTED_ADD (label);
+    //entry = katze_property_proxy (settings, "http-proxy", "address");
+    //SPANNED_ADD (entry);
+    //g_signal_connect (settings, "notify::proxy-type",
+    //    G_CALLBACK (midori_preferences_notify_proxy_type_cb), entry);
+    //midori_preferences_notify_proxy_type_cb (settings, NULL, entry);
+    //label = gtk_label_new (_("Port"));
+    //gtk_misc_set_alignment (GTK_MISC (label), 0.0, 0.5);
+    //INDENTED_ADD (label);
+    //entry = katze_property_proxy (settings, "http-proxy-port", NULL);
+    //SPANNED_ADD (entry);
+    //g_signal_connect (settings, "notify::proxy-type",
+    //    G_CALLBACK (midori_preferences_notify_proxy_type_cb), entry);
+    //midori_preferences_notify_proxy_type_cb (settings, NULL, entry);
+    //INDENTED_ADD (gtk_event_box_new ());
+    //label = gtk_label_new (NULL);
+    //GString* proxy_types = g_string_new (NULL);
+    //g_string_append (proxy_types, "<span size=\"smaller\">");
+    //g_string_append (proxy_types, _("Supported proxy types:"));
+    //g_string_append_c (proxy_types, ' ');
     /* http is not implemented as a GIO extension */
-    g_string_append (proxy_types, "http:// https://");
-    GIOExtensionPoint* proxy_extension = g_io_extension_point_lookup ("gio-proxy");
-    GList* proxies = g_io_extension_point_get_extensions (proxy_extension);
-    for (; proxies; proxies = g_list_next (proxies))
-        g_string_append_printf (proxy_types, " %s://", g_io_extension_get_name (proxies->data));
-    g_string_append (proxy_types, "</span>");
-    gtk_label_set_markup (GTK_LABEL (label), proxy_types->str);
-    g_string_free (proxy_types, TRUE);
-    SPANNED_ADD (label);
-    g_signal_connect (settings, "notify::proxy-type",
-        G_CALLBACK (midori_preferences_notify_proxy_type_cb), label);
-    midori_preferences_notify_proxy_type_cb (settings, NULL, label);
+    //g_string_append (proxy_types, "http:// https://");
+    //GIOExtensionPoint* proxy_extension = g_io_extension_point_lookup ("gio-proxy");
+    //GList* proxies = g_io_extension_point_get_extensions (proxy_extension);
+    //for (; proxies; proxies = g_list_next (proxies))
+    //    g_string_append_printf (proxy_types, " %s://", g_io_extension_get_name (proxies->data));
+    //g_string_append (proxy_types, "</span>");
+    //gtk_label_set_markup (GTK_LABEL (label), proxy_types->str);
+    //g_string_free (proxy_types, TRUE);
+    //SPANNED_ADD (label);
+    //g_signal_connect (settings, "notify::proxy-type",
+    //    G_CALLBACK (midori_preferences_notify_proxy_type_cb), label);
+    //midori_preferences_notify_proxy_type_cb (settings, NULL, label);
 #ifndef HAVE_WEBKIT2
     if (soup_session_get_feature (webkit_get_default_session (), SOUP_TYPE_CACHE))
     {
